### PR TITLE
Makes Borg Army Actually Scale with Player Count

### PR DESCRIPTION
**Pull request by JamieD1**
8 Borgs on 25 pop is kinda stupid lol

:cl:  
tweak: Makes Borg Army Actually Scale with Player Count
/:cl:

### CODE REVIEW

**Comment by Hislyna** [comment]
Me and the boys on 5 pop needing 8 borgs

File: code/game/gamemodes/objective.dm [hunk 1]
```diff
@@ -487,8 +487,13 @@ GLOBAL_LIST_EMPTY(objectives)
 
 /datum/objective/robot_army
 	name = "robot army"
-	explanation_text = "Have at least eight active cyborgs synced to you."
 	martyr_compatible = 0
+	var/number_of_borgs = 2
+
+/datum/objective/robot_army/New()
+	. = ..()
+	number_of_borgs = ceil((GLOB.joined_player_list.len / 10) + 2)
```
Comment (JamieD1) [inline code/game/gamemodes/objective.dm]:
```suggestion
	number_of_borgs = CEILING((GLOB.joined_player_list.len / 10) + 2)
```

File: code/game/gamemodes/objective.dm [hunk 2]
```diff
@@ -487,8 +487,13 @@ GLOBAL_LIST_EMPTY(objectives)
 
 /datum/objective/robot_army
 	name = "robot army"
-	explanation_text = "Have at least eight active cyborgs synced to you."
 	martyr_compatible = 0
+	var/number_of_borgs = 2
+
+/datum/objective/robot_army/New()
+	. = ..()
+	number_of_borgs = CEILING((GLOB.joined_player_list.len / 10) + 2)
```
Comment (JamieD1) [inline code/game/gamemodes/objective.dm]:
```suggestion
	number_of_borgs = ROUND_UP((GLOB.joined_player_list.len / 10) + 2)
```

Comment (ToasterBiome) [inline code/game/gamemodes/objective.dm]:
```suggestion
	number_of_borgs = CEILING((length(GLOB.joined_player_list) / 10) + 2)
```

length(x) is faster than x.len

**Review by ToasterBiome — DISMISSED** [review]

File: code/game/gamemodes/objective.dm [hunk 3]
```diff
@@ -487,8 +487,13 @@ GLOBAL_LIST_EMPTY(objectives)
 
 /datum/objective/robot_army
 	name = "robot army"
-	explanation_text = "Have at least eight active cyborgs synced to you."
 	martyr_compatible = 0
+	var/number_of_borgs = 2
+
+/datum/objective/robot_army/New()
+	. = ..()
+	number_of_borgs = ROUND_UP((GLOB.joined_player_list.len / 10) + 2)
```
Comment (JamieD1) [inline code/game/gamemodes/objective.dm]:
```suggestion
	number_of_borgs = ROUND_UP((length(GLOB.joined_player_list) / 10) + 2)
```

File: code/game/gamemodes/objective.dm [hunk 4]
```diff
@@ -502,7 +507,7 @@ GLOBAL_LIST_EMPTY(objectives)
 		for(var/mob/living/silicon/robot/R in A.connected_robots)
 			if(R.stat != DEAD)
 				counter++
-	return counter >= 8
+	return counter >= number of borgs
```
Comment (JamieD1) [inline code/game/gamemodes/objective.dm]:
```suggestion
	return counter >= number_of_borgs
```

**Comment by ReddicusDragon** [comment]
ah yes, me singlehandedly cranking out 8 bazillion borgs because the other roboticist vanished

**Review by Moltijoe — APPROVED** [review]

**Pull request merged**